Scripts/Runtime/Entities/Lifecycle/Migration/IMigrationObserver.cs

```diff
@@ -0,0 +1,28 @@
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Implement and register with <see cref="WorldEntityMigrationSystem"/> to receive a notification when
+    /// Entities are being migrated from one <see cref="World"/> to another. This will allow for scheduling jobs to
+    /// handle any custom migration for data that refers to <see cref="Entity"/>s but is not automatically handled by
+    /// Unity.
+    /// NOTE: The jobs that are scheduled will be completed immediately, but this allows for taking advantage of
+    /// multiple cores. 
+    /// </summary>
+    public interface IMigrationObserver
+    {
+        /// <summary>
+        /// Implement to handle any custom migration work.
+        /// </summary>
+        /// <param name="dependsOn">The <see cref="JobHandle"/> to wait on.</param>
+        /// <param name="destinationWorld">The <see cref="World"/> the entities are moving to.</param>
+        /// <param name="remapArray">The remapping array for Entities that were in this world and are moving to
+        /// the next World. See <see cref="EntityRemapUtility.EntityRemapInfo"/> and <see cref="EntityRemapUtility"/>
+        /// for more details if custom usage is needed.</param>
+        /// <returns>The <see cref="JobHandle"/> that represents all the custom migration work to do.</returns>
+        public JobHandle MigrateTo(JobHandle dependsOn, World destinationWorld, ref NativeArray<EntityRemapUtility.EntityRemapInfo> remapArray);
```

Comment:
I tried, but everything trickles down to:

`EntityRemapUtility.RemapEntity` which requires a `ref NativeArray<EntityRemapUtility.EntityRemapInfo>`

I can't pass a `ref NativeArray<EntityRemapUtility.EntityRemapInfo>.ReadOnly` into it. 

I could do some pointer magic to get the pointer from the ReadOnly and construct a new NativeArray with the same pointer but this function always happens in a hot loop so it's probably better to just leave it as is. 


I COULD however wrap this NativeArray in a struct so that you as the developer only get the struct and can't mess with the array and then at the hot loop location, it can access the NativeArray via it being internal. 

That would keep people from accidentally messing with the Array. Thoughts?

Reply:
Nah, too much work. ReadOnly just seemed like an easy win. People just shouldn't be stupid

Reply:
Cool, sounds good.